azurerm/resource_arm_app_service_source_control_token.go

```diff
@@ -0,0 +1,119 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/web/mgmt/2018-02-01/web"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/locks"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+var appServiceSourceControlTokenResourceName = "azurerm_app_service_source_control_token"
+
+func resourceArmAppServiceSourceControlToken() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAppServiceSourceControlTokenCreateOrUpdate,
+		Read:   resourceArmAppServiceSourceControlTokenRead,
+		Update: resourceArmAppServiceSourceControlTokenCreateOrUpdate,
+		Delete: schema.Noop,
```

Comment:
Should we implement delete as setting the token to "" ? It stands to reason that if the resource is deleted the data should be cleared out. WDYT?

Reply:
This sounds very nice. Many thanks. Will add this tomorrow.

Reply:
Have made an attempt to fix this. Not sure what the most explicit and readable way to do this is.